test/powershell/Host/ConsoleHost.Tests.ps1

```diff
@@ -257,6 +257,39 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
             & $powershell -noprofile -c ' ' | Should -BeNullOrEmpty
             $LASTEXITCODE | Should -Be 0
         }
+
+        It "MyInvocation.MyCommand.Name should contain script file name when executing script with -File argument" -TestCases @(
+            @{Filename = "test.ps1"},
+            @{Filename = "test-no-ext"}
+        ) {
+            param($Filename)
+            $testFilePath = Join-Path $testdrive $Filename
+            Set-Content -Path $testFilePath -Value '$MyInvocation.MyCommand.Name'
+            $observed = & $powershell -File $testFilePath -NoProfile -NoLogo
```

Comment:
There was [an intentional breaking change](https://github.com/PowerShell/PowerShell/pull/15859) in PowerShell 7.2 related to `powershell -File`: when running on Windows, a file without `.ps1` extension is not allowed to run using `pwsh -File`. So, the test won't work on Windows.